### PR TITLE
Adding Route for User Role Updation

### PR DESCRIPTION
**Pull request by JavidSumra**
*Have you read the [Contributing Guidelines](https://github.com/jessesquires/.github/blob/master/CONTRIBUTING.md)?*  Yes

Issue #45 

## Describe your changes
With POST request to /api/admin/user/:id will attempt to update the user's role to "admin," provided the user exists and currently has the role of "user." The route handles different scenarios, such as user not found, user already being an admin, or any other errors during the process.

image

### CODE REVIEW

**Review by anand-harsh — CHANGES_REQUESTED** [review]
Hi @JavidSumra , don't commit package-lock.json and package.json, just use

git add <file.js> instead of git add .

**Comment by JavidSumra** [comment]
Hey @anand-harsh I made Changes by Unstaging `package.json` and `package-lock.json` now you can Check API.

**Comment by anand-harsh** [comment]
> Hey @anand-harsh I made Changes by Unstaging `package.json` and `package-lock.json` now you can Check API.

Yes, check the other things need to be resolved, I have mentioned above

**Comment by anand-harsh** [comment]
hI @JavidSumra , ONCE UPDATE YOUR BRANCH

**Review by anand-harsh — CHANGES_REQUESTED** [review]
Update your branch to latest commit to resolve conflicts

File: routes/userRoutes.js [hunk 1]
```diff
@@ -28,11 +29,11 @@ router
   .get(isAuthenticated, updateProfilePicture);
 
 router.route("/forgetpassword").post(forgetPassword);
-router.route("/resetpassword/:token").put(resetPassword);
 router.route("/addtoplaylist").post(isAuthenticated, addToPlaylist);
-router.route("/removefromplaylist").delete(isAuthenticated, removeFromPlaylist );
-router.route("/me").delete(isAuthenticated, handleDeleteSingleUser );
-router.route("/admin/users/:id").delete(isAuthenticated, handleAdminDelete );
-
+router.route("/admin/user/:id").post(isAuthenticated, updateUserRole);
```
[the remark this answers is not in the document]
Comment (JavidSumra) [reply routes/userRoutes.js:33]: Hey I will do it by tonight.

**Comment by JavidSumra** [comment]
Hey @anand-harsh Please Checkout Now my PR.

**Comment by JavidSumra** [comment]
Hey @anand-harsh Shall I need to Create a New Branch?

**Comment by anand-harsh** [comment]
LGTM✨✨

**Pull request merged**